src/index.ts

```diff
@@ -51,7 +51,7 @@ export function decorate(
 }
 
 export function metadata<MetadataValue>(metadataKey: MetadataKey, metadataValue: MetadataValue) {
-  return function decorator(target: Function, propertyKey?: PropertyKey) {
+  return function decorator(target: Target, propertyKey?: PropertyKey) {
```

Comment:
This is the fix for #83 

It is checked by types in [these tests](https://github.com/abraham/reflection/blob/master/src/metadata.test.ts#L29-L41). They cause type errors without this change.